### DISCUSSION TITLE
Can we use graphiql-explorer with the new graphql-http ?

### DISCUSSION THREAD

**Discussion by ShueiYang** [graphql/graphql-http#38] [Q&A] [answered]
Hi, 
Since express-graphql is deprecated can we still use graphiql explorer with the new graphql-http as replacement?

**Comment by benjie** [comment]
Yes, GraphiQL Explorer should work against any valid GraphQL API, I don't think it actually cares about the transport at all - it's schema/editor only.

**Comment by ShueiYang** [comment]
Ok thanks, because I do a quick test in localhost with http://localhost:3000/graphql it give me an error  "message": "Missing query"

**Comment by enisdenjo** [comment, marked as answer]
I think the question is more whether the library _ships_ with GraphiQL integrated. @ShueiYang that it does not and it never will, graphql-http is a reference implementation of the GraphQL over HTTP specification and as such follows the specification strictly without any additional features ([see disclaimer in readme](https://github.com/graphql/graphql-http#only-graphql-over-http)).

You can of course intercept the incoming requests and render GraphiQL whenever you wish.

**Reply by d-volution** [reply]
> You can of course intercept the incoming requests and render GraphiQL

OK, but how to do that? Yes I am kind of a noob, but I would appreciate some hint or a weblink to a tutorial or whatever.
Do I have to write a whole GraphiQL renderer for myself?

Thanks in advance!

**Reply by ShueiYang** [reply]
I am also a beginner at GraphQL, all i can say is since the main goal from graphql-http is to do only what it should do, I simply use Postman to test some query since they support now graphql query. If you need more advanced feature like graphiQL you can for example just add and use graph-client to use graphiQL, there is also bigger library like Apollo-server which integrated their own Graphiql rendering. (Already link by enisdenjo in the readme)

**Reply by enisdenjo** [reply]
That is a fair point guys, let me see what I can do here. 🤔

**Reply by benjie** [reply]
You can also mount a pre-packaged GraphiQL into your server separate from graphql-http. [Ruru](https://www.npmjs.com/package/ruru) for example can be rendered via basically any Node server.

**Comment by AmirHosein-Zare** [comment]
Regarding graphiql's disclaimer, is there an alternative?
There is no solution?

**Reply by enisdenjo** [reply]
You can always use one of more feature-full servers, like [GraphQL Yoga](https://the-guild.dev/graphql/yoga-server). It offers a lot of things out-of-the box, things like subscriptions, caching, GraphiQL explorer, etc.

**Reply by ElginHarten** [reply]
Ran into the same problem, but it turned out to be straightforward to add back the same or equivalent functionality.  I opted to install playground instead of GraphiQL this time around.  This is how I do it using express:

`
import { createHandler } from 'graphql-http/lib/use/express';
const expressPlayground = require('graphql-playground-middleware-express').default

 app.use('/graphql', createHandler({ schema: schema }));
 app.get('/playground', expressPlayground({ endpoint: '/graphql' }));
`

This works for a simple hello world schema.  I haven't figured out how to connect with my neo4j driver.  If anyone can provide some hints, please share.  Hope this works for you.

**Reply by ElginHarten** [reply]
For anyone interested, the solution for hooking neo4j driver is to craft your createHandler statement like so:

`        app.use('/graphql', createHandler({
            schema: schema,
            context: (req) => ({ req, executionContext: driver })
         }));
`

**Comment by lars-thurman** [comment]


**Reply by enisdenjo** [reply]
Which tutorial?
